spec/rubocop/cop/team_spec.rb

```diff
@@ -368,13 +368,15 @@ def foo
 
     context 'when cop with different checksum joins' do
       before do
+        # rubocop:disable RSpec/LeakyConstantDeclaration
         module Test
           class CopWithExternalDeps < ::RuboCop::Cop::Cop
             def external_dependency_checksum
               'something other than nil'
             end
           end
         end
+        # rubocop:enable RSpec/LeakyConstantDeclaration
```

Comment:
A quick and dirty to overcome this inconvenience with defining cops in specs with `Class.new`:
```
diff --git a/lib/rubocop/cop/registry.rb b/lib/rubocop/cop/registry.rb
index 150dc29e0..e311b8fda 100644
--- a/lib/rubocop/cop/registry.rb
+++ b/lib/rubocop/cop/registry.rb
@@ -23,8 +23,7 @@ module RuboCop
     # Registry that tracks all cops by their badge and department.
     class Registry
       def initialize(cops = [], options = {})
-        @registry = {}
-        @departments = {}
+        @lazy = []
         @cops_by_cop_name = Hash.new { |hash, key| hash[key] = [] }
 
         cops.each { |cop| enlist(cop) }
@@ -32,10 +31,7 @@ module RuboCop
       end
 
       def enlist(cop)
-        @registry[cop.badge] = cop
-        @departments[cop.department] ||= []
-        @departments[cop.department] << cop
-        @cops_by_cop_name[cop.cop_name] << cop
+        @lazy << cop
       end
 
       # @return [Array<Symbol>] list of departments for current cops.
@@ -129,12 +125,28 @@ module RuboCop
         @cops_by_cop_name
       end
 
+      def registry
+        @registry ||= init_registry
+      end
+
+      def init_registry
+        @registry = {}
+        @departments = {}
+        @lazy.each do |cop|
+          @registry[cop.badge] = cop
+          @departments[cop.department] ||= []
+          @departments[cop.department] << cop
+          @cops_by_cop_name[cop.cop_name] << cop
+        end
+        @registry
+      end
+
       def cops
-        @registry.values
+        registry.values
       end
 
       def length
-        @registry.size
+        registry.size
       end
 
       def enabled(config, only, only_safe = false)
@@ -172,7 +184,7 @@ module RuboCop
       end
 
       def sort!
-        @registry = Hash[@registry.sort_by { |badge, _| badge.cop_name }]
+        @registry = Hash[registry.sort_by { |badge, _| badge.cop_name }]
 
         self
       end
@@ -214,7 +226,7 @@ module RuboCop
       end
 
       def registered?(badge)
-        @registry.key?(badge)
+        registry.key?(badge)
       end
     end
   end
```

Reply:
I understand that the only reason to change how it is done now is to make tests simpler/more conformant.

Reply:
Cool! Would you make that into a PR?

Reply:
Here you go https://github.com/rubocop-hq/rubocop/pull/8023